mutis/signal.py

```diff
@@ -109,35 +118,44 @@ def OU_fit(self, bins=None, rang=None, a=1e-5, b=100):
             # print('pcov: ')
             # print(np.sqrt(np.diag(pcov)))
             x_c = np.linspace(1e-5, 1.1 * np.max(x), 1000)
-            plt.plot(x_c, self.pdf(x_c, *popt), "k-", label="curve_fit", alpha=0.8)
-            res["curve_fit"] = (popt, np.sqrt(np.diag(pcov)))
+            plt.plot(x_c, self.pdf(x_c, *popt), 'k-', label='curve_fit', alpha=0.8)
+            res['curve_fit'] = (popt, np.sqrt(np.diag(pcov)))
         except Exception as e:
+            popt, pcov = None, None
             raise Exception(f"Some error fitting with curve_fit {e}")
 
+        # fit pdf with MLE
         # TODO: place outside as a helper class
-        #       and understand how it is used
-        #       is it mu really needed ?
         #       mu here is different than self.mu
-        # fit pdf with MLE
         class OU(scipy_stats.rv_continuous):
             def _pdf(self, x, l, mu):
                 return (l * mu) ** (1 + l) / scipy_special.gamma(1 + l) * np.exp(-l * mu / x) / x ** (l + 2)
 
         try:
-            fit = OU(a=a, b=np.percentile(self.values, b)).fit(self.values, 1, 1, floc=0, fscale=1)
+            fit = OU(a=a, b=np.percentile(y, b)).fit(y, 1, 1, floc=0, fscale=1)
             # print('MLE fit: (l, mu)')
             # print(fit)
             x_c = np.linspace(0, 1.1 * np.max(x), 1000)
-            plt.plot(x_c, self.pdf(x_c, fit[0], fit[1]), "k-.", label="MLE", alpha=0.8)
-            res["MLE_fit"] = fit[:-2]
+            plt.plot(x_c, self.pdf(x_c, fit[0], fit[1]), 'k-.', label='MLE', alpha=0.8)
+            res['MLE_fit'] = fit[:-2]
         except Exception as e:
             raise Exception(f"Some error fitting with MLE {e}")
+
         plt.legend(loc="lower right")
+
         plt.show()
 
-        # estimate theta
-        res["th_est1"] = fit[0] * sigma_est ** 2 / 2
-        res["th_est2"] = popt[0] * sigma_est ** 2 / 2
+        # estimate theta (from curve_fit)
+        try:
+            res['th_est1'] = fit[0] * sigma_est ** 2 / 2
+        except:
+            res['th_est1'] = None
+
+        # estimate theta (from MLE)
+        try:
+            res['th_est2'] = popt[0] * sigma_est ** 2 / 2
+        except:
+            res['th_est2'] = None
```

Comment:
They are there to ensure the program returns the rest of information even if `popt`, `pcov` do not exist, which happens sometimes when curve_fit fails, but MLE does not.

Reply:
except blocks here will be never reached because there are exceptions raised above if the process to get `fit`, `popt` or `sigma_est` fails, so why adding these try/except blocks here?

Reply:
These exceptions are reached even if computing `popt` or `fits` fail, because they are also inside exceptions. These exceptions are here so the maximum information is returned even if some computations fail (see https://github.com/IAA-CSIC/MUTIS/pull/18#discussion_r639761787 and https://github.com/IAA-CSIC/MUTIS/pull/18#issuecomment-848809065 point 3)

Reply:
(I edited the comment above)